internal/ignition/ignition.go

```diff
@@ -445,6 +458,38 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte,
 	return nil
 }
 
+func (g *installerGenerator) applyInfrastructureCRPatch(ctx context.Context) error {
+	log := logutil.FromContext(ctx, g.log)
+
+	// We are only patching the InfrastructureCR if the hosts count is 4
+	// and the three masters are schedulable.
+	if len(g.cluster.Hosts) != 4 {
+		log.Debugf("number of hosts is different than 4, no need to patch the Infrastructure CR %s", len(g.cluster.Hosts))
+		return nil
+	}
+
```

Comment:
Can we have log line like "Patching ...." i think it will be useful for debugging

Reply:
I will add it! :+1: